ccm_web/server/src/api/dtos/api.section.external.users.dto.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  ArrayMaxSize,
+  ArrayMinSize,
+  IsArray,
+  IsNotEmpty,
+  ValidateNested
+} from 'class-validator'
+import { ApiProperty } from '@nestjs/swagger'
+import { Type } from 'class-transformer'
+import { UserEnrollmentType } from '../../canvas/canvas.interfaces'
+
+export class SectionExternalUserDto {
+  @ApiProperty()
+  @IsNotEmpty()
+  email: string
+
+  @ApiProperty()
+  @IsNotEmpty()
+  givenName: string
+
+  @ApiProperty()
+  @IsNotEmpty()
+  surname: string
+
+  @ApiProperty({ enum: UserEnrollmentType })
+  @IsNotEmpty()
+  type: UserEnrollmentType
```

Comment:
I think we can make these more strict. I think the first three should also have maximum lengths of 255 (I'm guessing that's what the Canvas restriction is).

For the enrollment, I noticed that the existing enrollment endpoint also doesn't seem to validate that the entry is a member in the enum. I think there are some additional `class-validator` options we can explore to do this kind of thing as well.

Reply:
I'm adding ` @MaxLength(255)` annotations for the strings.

I'm not sure how to implement a restriction on the enrollment type.  Is something similar being done in the UI to validate types specified in the uploaded CSVs?

Reply:
Yes, we check whether the roles are valid when submitted through CSV.

Here I think we could probably use `@IsIn` from `class-validator` and try passing `Object.values(UserEnrollmentType)`? See common validation decorators [here](https://github.com/typestack/class-validator#validation-decorators).

Reply:
Just FYI, I'm working on a separate issue, and this seems to be working for me, in terms of validating whether the role is in the controlled vocabulary (concretized by the enum `UserEnrollmentType`).

https://github.com/ssciolla/canvas-course-manager-next/blob/599f16693f4b292b28ce8e6f424c3eac347dda88/ccm_web/server/src/api/dtos/api.section.enrollment.dto.ts#L21-L24

Reply:
No longer relevant to this PR. I believe `@IsIn(Object.values(UserEnrollmentType))` is in place for the enrollment endpoints now following changes in other PRs.